### DISCUSSION TITLE
Extending sidebar with custom groups and elements

### DISCUSSION THREAD

**Discussion by SebastianBienert** [scalar/scalar#781] [Q&A]
Hi guys,

I would like to extend a sidebar generated by the regular ApiReference component with additional static groups of items (like documentation overview etc.). These categories would not be directly connected to OpenAPI reference. I thought I could use SidebarElement and SidebarGroup components to achieve this by adding them on top of the existing sidebar but these are not exported from the library.

<img width="897" alt="image" src="https://github.com/scalar/scalar/assets/17458785/180fcf75-e96c-4983-b229-f9f13666c6fa">

Is there any other way to extend the sidebar element list by custom elements? Is there a chance to export [SidebarGroup ](https://github.com/scalar/scalar/blob/main/packages/api-reference/src/components/SidebarGroup.vue)and [SidebarElement ](https://github.com/scalar/scalar/blob/main/packages/api-reference/src/components/SidebarElement.vue)components from '@scalar/api-reference' so developers could use them within their implementation?

<img width="482" alt="image" src="https://github.com/scalar/scalar/assets/17458785/1072329a-bfe3-4bed-89f3-87faf7e68a46">

Asa a reference I am attaching question from RapiDoc library:
[https://github.com/rapi-doc/RapiDoc/issues/633](https://github.com/rapi-doc/RapiDoc/issues/633)
and their solution:
[https://rapidocweb.com/examples/markdown-headings.html#overview--documentation](https://rapidocweb.com/examples/markdown-headings.html#overview--documentation)

**Comment by hanspagel** [comment]
Great idea! What you can do already: Add Markdown to your description, similar to what RapiDoc has. See an example here:

https://github.com/outline/openapi/blob/main/spec3.json

Headings will be added to the sidebar.
